plugin/advisor/advisor.go

```diff
@@ -107,6 +108,29 @@ type Advice struct {
 	Content string
 }
 
+// MarshalLogObject constructs a field that carries Advice.
+func (a Advice) MarshalLogObject(enc zapcore.ObjectEncoder) error {
+	enc.AddString("status", a.Status.String())
+	enc.AddInt("code", int(a.Code))
+	enc.AddString("title", a.Title)
+	enc.AddString("content", a.Content)
+	return nil
+}
+
+// AdviceList is a slice of Advice.
+// TODO(rebelice): use AdviceList instead of []Advice
+type AdviceList []Advice
```

Comment:
I think the `AdviceList` type looks a bit redundant. `[]Advice` is enough.

Reply:
I agree with you. I added `AdviceList` to use the zap logger, such as
```go
log.Debug("Query result",
  zap.String("statement", exec.Statement),
  zap.String("data", resultSet.Data),
  zap.Array("advice", resultSet.Advice),
)
```

Do you know a better way to log the advice list without the `AdviceList` type?